matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/verification/VerificationTransportRoomMessage.kt

```diff
@@ -1,382 +1,316 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.matrix.android.sdk.internal.crypto.verification
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.launch
+import org.matrix.android.sdk.api.session.crypto.verification.CancelCode
+import org.matrix.android.sdk.api.session.crypto.verification.ValidVerificationInfoRequest
+import org.matrix.android.sdk.api.session.crypto.verification.VerificationTxState
+import org.matrix.android.sdk.api.session.events.model.Content
+import org.matrix.android.sdk.api.session.events.model.Event
+import org.matrix.android.sdk.api.session.events.model.EventType
+import org.matrix.android.sdk.api.session.events.model.LocalEcho
+import org.matrix.android.sdk.api.session.events.model.RelationType
+import org.matrix.android.sdk.api.session.events.model.UnsignedData
+import org.matrix.android.sdk.api.session.events.model.toContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageVerificationAcceptContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageVerificationCancelContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageVerificationDoneContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageVerificationKeyContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageVerificationMacContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageVerificationReadyContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageVerificationRequestContent
+import org.matrix.android.sdk.api.session.room.model.message.MessageVerificationStartContent
+import org.matrix.android.sdk.api.session.room.model.relation.RelationDefaultContent
+import org.matrix.android.sdk.internal.crypto.model.rest.VERIFICATION_METHOD_RECIPROCATE
+import org.matrix.android.sdk.internal.crypto.model.rest.VERIFICATION_METHOD_SAS
+import org.matrix.android.sdk.internal.crypto.tasks.SendVerificationMessageTask
+import org.matrix.android.sdk.internal.session.room.send.LocalEchoEventFactory
+import org.matrix.android.sdk.internal.task.SemaphoreCoroutineSequencer
+import timber.log.Timber
+import java.util.concurrent.Executors
+
+internal class VerificationTransportRoomMessage(
+        private val sendVerificationMessageTask: SendVerificationMessageTask,
+        private val userId: String,
+        private val userDeviceId: String?,
+        private val roomId: String,
+        private val localEchoEventFactory: LocalEchoEventFactory,
+        private val tx: DefaultVerificationTransaction?
+) : VerificationTransport {
+
+    private val dispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
+    private val verificationSenderScope = CoroutineScope(SupervisorJob() + dispatcher)

```

Comment:
What will happen if the session is signed out? I am afraid this scope will not be cancelled.

Reply:
There should never be a lot of pending task in that scope. Not sure if I should invest more on that part.
Do you think it's acceptable to leave it like that?

Reply:
Ok. I have made the verification scope as a child of the cryptoscope. So when the cryptoService will be closed it will cancel all jobs in the transport

Reply:
Thanks!